src/microbe_stage/MicrobeVisualOnlySimulation.cs

```diff
@@ -242,15 +242,29 @@ public Vector3 CalculateMicrobePhotographDistance()
         // Calculate cell center graphics position for more accurate photographing
         if (organelles.CreatedOrganelleVisuals is { Count: > 0 })
         {
-            foreach (var node in organelles.CreatedOrganelleVisuals.Values)
+            foreach (var pair in organelles.CreatedOrganelleVisuals)
             {
+                // We don't need to account for internal organelles as they are located within the cell's radius
+                if (!pair.Key.Definition.PositionedExternally)
+                    continue;
+
                 // TODO: is there another way to not need to call so many Godot data access methods here
                 // Organelle positions might be usable as the visual positions are derived from them, but this requires
                 // using the global translation for some reason as translation gives just 0 here and doesn't help.
-                center += node.GlobalPosition;
-            }
 
-            center /= organelles.CreatedOrganelleVisuals.Count;
+                // Assume that the organelle's radius is 1
+                float organelleRadius = 1.0f;
+                float distance = (pair.Value.GlobalPosition - center).Length();
```

Comment:
Would it be possible to use squared length here? That way the if could discard many organelles and only a few sqrt calls would be needed for the organelles that actually need to push the radius.